src/primitive.rs

```diff
@@ -0,0 +1,213 @@
+// derivative works of, display, perform, and distribute the Software and make,
+// use, sell, offer for sale, import, export, have made, and have sold the
+// Software and the Larger Work(s), and to sublicense the foregoing rights on
+// either these or other terms.
+//
+// This license is subject to the following condition: The above copyright
+// notice and either this complete permission notice or at a minimum a reference
+// to the UPL must be included in all copies or substantial portions of the
+// Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+use byteorder::{ByteOrder, LittleEndian};
+
+use rustc_const_math::ConstInt;
+use rustc::mir::{Operand, Constant, Literal};
+use rustc::middle::const_val::ConstVal;
+use rustc::ty::{Ty, TypeVariants};
+use syntax::ast::{IntTy, UintTy};
+
+use interp::TypedVal;
+use machine::Machine;
+
+/// An operand's value can always be determined without needing to temporarily
+/// save values to a stack. This is because of the recursive nature of the MIR.
+/// An operand will only ever return a primitive value or a pointer.
+pub fn eval_operand<'a,'tcx>(operand: &Operand<'tcx>,
+                             vm: &mut Machine<'a, 'tcx>)
+                             -> TypedVal<'tcx> {
+    let ty = operand.ty(vm.cur_frame().mir, vm.tcx);
+    match *operand {
+        Operand::Copy(ref place) |
+        Operand::Move(ref place) => {
+            unimplemented!()
+        }
+        Operand::Constant(ref constant) => {
+            let Constant {ref literal, ..} = **constant;
+            let val = match *literal {
+                Literal::Value { ref value } => value.val.to_bytes(),
+                Literal::Promoted { index } => {
+                    unimplemented!()
+                }
+            };
+            return TypedVal {
+                val,
+                ty
+            }
+        }
+    }
+}
+
+trait ByteCast {
+    type T;
+
+    fn to_bytes(&self) -> Vec<u8>;
+
+    fn new_from(bytes: &[u8], ty: TypeVariants) -> Self::T;
+}
+
+impl<'tcx> ByteCast for ConstVal<'tcx> {
+    type T = ConstVal<'tcx>;
+
+    fn to_bytes(&self) -> Vec<u8> {
+        match *self {
+            ConstVal::Integral(ref ci) => ci.to_bytes(),
+            _ => unimplemented!()
+        }
+    }
+
+    fn new_from(bytes: &[u8], ty: TypeVariants)-> ConstVal<'tcx> {
+        unimplemented!()
+    }
+}
+
+impl<'tcx> ByteCast for ConstInt {
+    type T = ConstInt;
+
+    fn to_bytes(&self) -> Vec<u8> {
+        match *self {
```

Comment:
Although I know the syntax doesn't require it (why not I don't know!), I think it's better to use `,` after every match arm for clarity.

Reply:
Okay makes sense. Will do!

Reply:
Also in favour of trailing commas.